dashboard/app/models/level_source.rb

```diff
@@ -43,6 +43,41 @@ def self.cache_key(level_id, md5)
     "#{level_id}-#{md5}"
   end
 
+  # Returns an obfuscated ID (URL) from which the LevelSource can be accessed.
+  # @param [Integer] user_id The ID of the user to obfuscate the LevelSource ID with. Note that the
+  #   user_id is a source for obfuscation only, not authorization or authentication.
+  # @return [String] The /c/ link, incorporating the user ID and the level source ID.
+  def c_link_from_user_id(user_id)
+    raise unless user_id
+    Base64.urlsafe_encode64 "#{id}:#{user_id}"
+  end
+
+  # Returns the LevelSource ID associated with an unobfuscated or obfuscated /c/ link. If the link
+  # is obfuscated and verify_user is true, returns nil if the obfuscated user does not exist and in various edge (error) cases.
+  # @param [String] c_link The /c/ link, unobfuscated or obfuscated.
+  # @param [Boolean] verify_user Whether the user's existence should be verified.
+  # @return [Integer, nil] The associated LevelSource ID.
+  def self.level_source_id_from_c_link(c_link, verify_user:)
+    # The /c/ link is not obfuscated, so simply return it.
+    return c_link.to_i if c_link == c_link.to_i.to_s
```

Comment:
you're assuming that the result of base64-encoding "#{id}:#{user_id}" will always be non-numerical. The correctness of this assumption is non-obvious, however it appears to be true empirically with inputs of this form. I would recommend adding a comment stating that we're assuming this is true for inputs containing only `0`-`9` and `:`.